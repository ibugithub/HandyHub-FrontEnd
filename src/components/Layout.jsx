import React from 'react';
import { Outlet } from 'react-router-dom';
import Navbar from './Navbar';

const Layout = () => (
  <div style={{ display: 'flex', height: '100vh' }}>
    <Navbar />
    <div
      className="mx-auto w-full h-full bg-gray-100"
      style={{
        flex: 1, padding: '30px 20px 20px', height: '100%', overflowY: 'auto',
      }}
    >
      <Outlet />
    </div>
  </div>
);

export default Layout;
