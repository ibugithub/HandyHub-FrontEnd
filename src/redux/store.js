import { configureStore } from '@reduxjs/toolkit';
import tradesReducer from './reducers/tradesSlice';
import tradeDetailsReducer from './reducers/tradeDetailsSlice';
import { logoutReducer } from './reducers/auth/logoutSlice';
import { registerReducer } from './reducers/auth/registerSlice';
import { loginReducer } from './reducers/auth/loginSlice';
import { reserveReducer } from './reducers/resereveSlice';
import { reservationReducer } from './reducers/rservationSlice';

const store = configureStore({
  reducer: {
    trades: tradesReducer,
    tradeDetails: tradeDetailsReducer,
    register: registerReducer,
    login: loginReducer,
    logout: logoutReducer,
    reservations: reservationReducer,
    reserve: reserveReducer,
  },
});

export default store;
