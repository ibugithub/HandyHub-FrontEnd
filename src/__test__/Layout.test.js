import '@testing-library/jest-dom';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Layout from '../components/Layout';

describe('Testing the Layout Component', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Layout />
        </Router>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
