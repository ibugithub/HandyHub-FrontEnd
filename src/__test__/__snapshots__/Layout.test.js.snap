// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout Component matches snapshot 1`] = `
<div>
  <div
    style="display: flex; height: 100vh;"
  >
    <button
      aria-controls="default-sidebar"
      class="items-start p-2 mt-2 ml-3 text-sm text-gray-500 rounded-lg sm:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
      data-drawer-target="default-sidebar"
      data-drawer-toggle="default-sidebar"
      style="position: absolute; margin: 0px;"
      type="button"
    >
      <span
        class="sr-only"
      >
        Open sidebar
      </span>
      <svg
        aria-hidden="true"
        class="w-6 h-6"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M2 4.75A.75.75 0 012.75 4h14.5a.75.75 0 010 1.5H2.75A.75.75 0 012 4.75zm0 10.5a.75.75 0 01.75-.75h7.5a.75.75 0 010 1.5h-7.5a.75.75 0 01-.75-.75zM2 10a.75.75 0 01.75-.75h14.5a.75.75 0 010 1.5H2.75A.75.75 0 012 10z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
    <aside
      aria-label="Sidebar"
      class="fixed top-0 left-0 z-40 w-full h-screen bg-white transition-transform sm:relative sm:w-64 -translate-x-full sm:translate-x-0"
      id="default-sidebar"
    >
      <nav
        class="h-screen flex flex-col justify-between items-center border-e-2 p-3"
      >
        <h1
          class="text-3xl mt-4 -rotate-6 text-bold underline"
          style="font-family: Dancing Script;"
        >
          HandyHome
        </h1>
        <div
          class="flex flex-col justify-center text-gray-700 py-8"
        >
          <div
            class="footer-toolbar flex justify-center space-x-4 text-gray-700"
          >
            <a
              class="text-sm"
              href="https://twitter.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-twitter text-gray-700 text-sm"
                data-icon="twitter"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              class="text-sm"
              href="https://facebook.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-facebook text-gray-700 text-sm"
                data-icon="facebook"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              class="text-sm"
              href="https://plus.google.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-google-plus text-gray-700 text-sm"
                data-icon="google-plus"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256,8C119.1,8,8,119.1,8,256S119.1,504,256,504,504,392.9,504,256,392.9,8,256,8ZM185.3,380a124,124,0,0,1,0-248c31.3,0,60.1,11,83,32.3l-33.6,32.6c-13.2-12.9-31.3-19.1-49.4-19.1-42.9,0-77.2,35.5-77.2,78.1S142.3,334,185.3,334c32.6,0,64.9-19.1,70.1-53.3H185.3V238.1H302.2a109.2,109.2,0,0,1,1.9,20.7c0,70.8-47.5,121.2-118.8,121.2ZM415.5,273.8v35.5H380V273.8H344.5V238.3H380V202.8h35.5v35.5h35.2v35.5Z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              class="text-sm"
              href="https://github.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-github text-gray-700 text-sm"
                data-icon="github"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              class="text-sm"
              href="https://pinterest.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-pinterest text-gray-700 text-sm"
                data-icon="pinterest"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M496 256c0 137-111 248-248 248-25.6 0-50.2-3.9-73.4-11.1 10.1-16.5 25.2-43.5 30.8-65 3-11.6 15.4-59 15.4-59 8.1 15.4 31.7 28.5 56.8 28.5 74.8 0 128.7-68.8 128.7-154.3 0-81.9-66.9-143.2-152.9-143.2-107 0-163.9 71.8-163.9 150.1 0 36.4 19.4 81.7 50.3 96.1 4.7 2.2 7.2 1.2 8.3-3.3.8-3.4 5-20.3 6.9-28.1.6-2.5.3-4.7-1.7-7.1-10.1-12.5-18.3-35.3-18.3-56.6 0-54.7 41.4-107.6 112-107.6 60.9 0 103.6 41.5 103.6 100.9 0 67.1-33.9 113.6-78 113.6-24.3 0-42.6-20.1-36.7-44.8 7-29.5 20.5-61.3 20.5-82.6 0-19-10.2-34.9-31.4-34.9-24.9 0-44.9 25.7-44.9 60.2 0 22 7.4 36.8 7.4 36.8s-24.5 103.8-29 123.2c-5 21.4-3 51.6-.9 71.2C65.4 450.9 0 361.1 0 256 0 119 111 8 248 8s248 111 248 248z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
          <p
            class="text-center text-sm text-gray-700 pt-2"
          >
            Â© 2023 by HandyHome
          </p>
        </div>
      </nav>
    </aside>
    <div
      class="mx-auto w-full h-full bg-gray-100"
      style="flex: 1; padding: 30px 20px 20px; height: 100%; overflow-y: auto;"
    />
  </div>
</div>
`;
